app/javascript/vue/repository_item_sidebar/repository_values/RepositoryNumberValue.vue

```diff
@@ -1,48 +1,102 @@
 <template>
-  <div id="repository-number-value-wrapper" class="flex flex-col min-min-h-[46px] h-auto gap-[6px]">
+  <div id="repository-number-value-wrapper"
+       class="flex flex-col min-h-[46px] h-auto gap-[6px]"
+  >
     <div class="font-inter text-sm font-semibold leading-5 flex justify-between">
       <div class="truncate w-4/5">{{ colName }}</div>
-      <div @click="toggleExpandContent" v-show="expendable" class="font-normal leading-5 btn-text-link">
-        {{ this.contentExpanded ? i18n.t('repositories.item_card.repository_number_value.collapse') :
-          i18n.t('repositories.item_card.repository_number_value.expand') }}
+      <div @click="toggleCollapse"
+           v-show="expandable"
+           class="font-normal leading-5 btn-text-link">
+        {{
+          collapsed
+            ? i18n.t("repositories.item_card.repository_number_value.expand")
+            : i18n.t("repositories.item_card.repository_number_value.collapse")
+        }}
       </div>
     </div>
-    <div v-if="colVal" ref="numberRef"
-      class="text-sn-dark-grey font-inter text-sm font-normal leading-5 min-h-[20px] overflow-y-auto"
-      :class="{ 'max-h-[60px]': !contentExpanded, 'max-h-[600px]': contentExpanded }">
-      {{ colVal }}
-    </div>
-    <div v-else class="text-sn-dark-grey font-inter text-sm font-normal leading-5">
-      {{ i18n.t('repositories.item_card.repository_number_value.no_number') }}
+    <div>
+      <inline-edit
+        v-if="permissions?.can_manage_repository_rows && !inArchivedRepository"
+        ref="numberRef"
+        :value="colVal"
+        :placeholder="i18n.t('repositories.item_card.repository_number_value.placeholder')"
+        :noContentPlaceholder="i18n.t('repositories.item_card.repository_number_value.no_number')"
+        :allowBlank="true"
+        :smartAnnotation="false"
+        :attributeName="`repository-number-value ${colId}`"
```

Comment:
some strange name

Reply:
Should probably be something more generic and readable, this is used for displaying blank field errors and such. In the example if you have a protocol name, the attributeName would be `:attributeName="`${i18n.t('Protocol')} ${i18n.t('name')}`"`, so it can say "Protocol name can't be blank", for example.